crates/net/downloaders/src/bodies/bodies.rs

```diff
@@ -279,7 +279,9 @@ where
                     _ => Range::default(),
                 };
 
-                let mut requests = Vec::<RangeInclusive<BlockNumber>>::default();
+                let max_request_len = self.request_limit.min(self.stream_batch_size as u64);
```

Comment:
this is different than the logic used in `next_headers_request` up above:

```
        let limit = self.download_range.end.saturating_sub(start_at).min(self.request_limit);
```

using the same logic also seems to simplify L297 in this PR

Reply:
on range recovery, we simplify the logic with the aforementioned line, because `query_headers` can return less headers than the requested range (in the range exceeds the stream batch size)